apps/test/unit/templates/rubrics/RubricFloatingActionButtonTest.jsx

```diff
@@ -134,68 +112,93 @@ describe('RubricFloatingActionButton - React Testing Library', () => {
-    const wrapper = shallow(<RubricFloatingActionButton {...defaultProps} />);
-    expect(wrapper.find('button').length).to.equal(1);
-    wrapper.find('button').simulate('click');
-    expect(wrapper.find('Connect(RubricContainer)').length).to.equal(1);
+    render(
+      <Provider store={store}>
+        <RubricFloatingActionButton {...defaultProps} />
+      </Provider>
+    );
+    const button = screen.getByRole('button', {
+      name: i18n.openOrCloseTeachingAssistant(),
+    });
+    fireEvent.click(button);
+    expect(screen.getByText(i18n.rubricAiHeaderText())).toBeVisible();
   });
 
   it('sends events when opened and closed', () => {
-    const sendEventSpy = sinon.stub(analyticsReporter, 'sendEvent');
     const reportingData = {unitName: 'test-2023', levelName: 'test-level'};
-    const wrapper = shallow(
-      <RubricFloatingActionButton
-        {...defaultProps}
-        reportingData={reportingData}
-      />
+    render(
+      <Provider store={store}>
+        <RubricFloatingActionButton
+          {...defaultProps}
+          reportingData={reportingData}
+        />
+      </Provider>
     );
-    wrapper.find('button').simulate('click');
-    expect(sendEventSpy).to.have.been.calledWith(
+    expect(screen.queryByText(i18n.rubricAiHeaderText())).not.toBeVisible();
+    const button = screen.getByRole('button', {
+      name: i18n.openOrCloseTeachingAssistant(),
+    });
+    fireEvent.click(button);
+    expect(screen.getByText(i18n.rubricAiHeaderText())).toBeVisible();
+
+    expect(sendEventSpy).toHaveBeenCalledWith(
       EVENTS.TA_RUBRIC_OPENED_FROM_FAB_EVENT,
       {
         ...reportingData,
         viewingStudentWork: false,
         viewingEvaluationLevel: true,
       }
     );
-    wrapper.find('button').simulate('click');
-    expect(sendEventSpy).to.have.been.calledWith(
+
+    fireEvent.click(button);
+    expect(screen.queryByText(i18n.rubricAiHeaderText())).not.toBeVisible();
+
+    expect(sendEventSpy).toHaveBeenCalledWith(
       EVENTS.TA_RUBRIC_CLOSED_FROM_FAB_EVENT,
       {
         ...reportingData,
         viewingStudentWork: false,
         viewingEvaluationLevel: true,
       }
     );
-    sendEventSpy.restore();
+  });
+
+  it('sends open on page load event when open state is true in session storage', () => {
+    sessionStorage.setItem('RubricFabOpenStateKey', 'true');
+    render(
+      <Provider store={store}>
+        <RubricFloatingActionButton {...defaultProps} />
+      </Provider>
+    );
+    expect(sendEventSpy).toHaveBeenCalledWith(
+      EVENTS.TA_RUBRIC_OPEN_ON_PAGE_LOAD,
+      {
+        viewingStudentWork: false,
+        viewingEvaluationLevel: true,
+      }
+    );
+    const image = screen.getByRole('img', {name: 'AI bot'});
+    fireEvent.load(image);
+    const fab = screen.getByRole('button', {
+      name: i18n.openOrCloseTeachingAssistant(),
+    });
+    expect(fab.classList.contains('unittest-fab-pulse')).toBe(false);
```

Comment:
this test case was moved out of the 'pulse animations' block. to see a diff of what changed in it: https://github.com/code-dot-org/code-dot-org/pull/62159/files/b0dc23ae46a6c174552c4c98dfc365849000fa49